test/src/userobjects/InterfaceValueUO_QP.C

```diff
@@ -0,0 +1,97 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceValueUO_QP.h"
+#include "MooseMesh.h"
+registerMooseObject("MooseTestApp", InterfaceValueUO_QP);
+
+template <>
+InputParameters
+validParams<InterfaceValueUO_QP>()
+{
+  InputParameters params = validParams<InterfaceValueUserObject>();
+  params.addRequiredCoupledVar("var", "The variable name");
+  params.addCoupledVar("var_neighbor", "The variable name");
+  params.addClassDescription("Test Interfae User Object computing and storing average values at "
+                             "each QP across an interface");
+  return params;
+}
+
+InterfaceValueUO_QP::InterfaceValueUO_QP(const InputParameters & parameters)
```

Comment:
Let's name this `InterfaceQpValueUserObject` as we don't traditionally use underscores in our file names and if we're going to reference the type of the object e.g. UserObject, we traditionally spell it out and put it as the suffix